04_設計/04_プラハチャレンジをDDDで実装してみる/02_APIサーバ/src/usecase/remove-member-from-pair.ts

```diff
@@ -0,0 +1,48 @@
+import { Identifier } from "domain/__shared__/identifier";
+import { PairFactory } from "domain/team/service/pair-factory";
+import { TeamFactory } from "domain/team/service/team-factory";
+import { ITeamRepository } from "domain/team/team-repository-interface";
+
+interface IExecuteProps {
+  memberID: string;
+  pairID: string;
+}
+export class RemoveMemberFromPair {
+  private readonly teamRepository: ITeamRepository;
+
+  public constructor(teamRepository: ITeamRepository) {
+    this.teamRepository = teamRepository;
+  }
+
+  public execute = async (props: IExecuteProps): Promise<void> => {
+    const currentTeam = await this.teamRepository.getByPairID({
+      pairID: new Identifier(props.pairID),
+    });
+
+    if (currentTeam === null) {
+      throw new Error("Pair is not exists");
+    }
+
+    const replacedPairList = currentTeam.pairList.map((currentPair) =>
+      currentPair.memberList.some(
+        (member) => member.id.value === props.memberID,
```

Comment:
```suggestion
        (member) => member.id.equals(new Identifer(props.memberID),
```
id.valueを使うと内部知識が漏れてしまっているように感じるので、上記のように識別子は識別子として比較する方が良いかと思います！